enhancements/oc/must-gather-server-side-configuration.md

```diff
@@ -0,0 +1,397 @@
+   checked by the linter.~~
+1. ~~**Fill out the "overview" sections.** This includes the Summary and
+   Motivation sections. These should be easy and explain why the community
+   should desire this enhancement.~~
+1. **Create a PR.** Assign it to folks with expertise in that domain to help
+   sponsor the process.
+1. **Merge after reaching consensus.** Merge when there is consensus
+   that the design is complete and all reviewer questions have been
+   answered so that work can begin.  Come back and update the document
+   if important details (API field names, workflow, etc.) change
+   during code review.
+1. **Keep all required headers.** If a section does not apply to an
+   enhancement, explain why but do not remove the section. This part
+   of the process is enforced by the linter CI job.
+
+See ../README.md for background behind these instructions.
+
+Start by filling out the header with the metadata for this enhancement.
+
+# Cluster side configuration for must-gather
+
+## Summary
+
+The idea is to allow configuration of must gather via an object deployed to
+a cluster. This allows better match to specific data collection needs based
+on the purpose of the cluster and avoids human mistakes when running must-gather
+manually.
+
+## Motivation
+
+Many customer cases and bug reports I have been exposed to lack the proper
+data needed for debugging the cluster at hand.
+
+1. Sometimes it is caused by humans not following the right procedure
+   (such as adding an extra must-gather image to the command line),
+1. in other cases it was caused by special needs that must gather by default
+   cannot satisfy for scale reasons (eg. low level kernel details needed to
+   debug Telco RAN latency on single node clusters).
+
+### User Stories
+
+* As a cluster administrator (owner) I want to be able to force all must-gather
+  executions to collect the necessary data for debugging of this specific
+  cluster.
+* As a cluster operator I want to use simple `oc adm must-gather` and collect
+  all data necessary without having to re-read multiple documentation pages.
+  In other words, I want the command to be as simple as possible with no
+  complicated command line arguments needed.
+* As an application and cluster owner I want a place to declare application
+  specific data collection needs so the probability of mistake when collecting
+  the debug snapshot is minimized.
+* As a cluster operator I want to collect only what is necessary to
+  minimize the collection time and the cluster load caused by the
+  collection tooling.
+* As a system add-on developer I want to be able to add extra configuration
+  for must-gather related to my add-on to collect the proper data. And I want
+  this even for add-ons that are NOT operators and have no active controllers
+  running.
+* As a system add-on developer I want to be able to tweak the behavior
+  of my must-gather image per-cluster.
```

Comment:
The logic to behave differently based on cluster contents could be encapsulated in the image. Can you give an example where this would not make sense?